### ISSUE TITLE
Prompt for registering as default atom:// protocol handler is unclear

### ISSUE THREAD

**Issue by carpben** [atom/atom#16453]
### Description
This is not a code issue, but an UX issue. I personally find this message: 
![image](https://user-images.githubusercontent.com/13496178/34385155-62980656-eb29-11e7-98e3-9eb85af46cb9.png)
unclear. What is a URI handler and why should we now set it on Atom. If you can add an explanation, or an explenation button it would be helpful. 

### Versions
1.23.1 X64
You can get this information from copy and pasting the output of `atom --version` and `apm --version` from the command line. Also, please include the OS and what version of the OS you're running.

**Comment by JohnLeicht** [comment]
I, too, find this unclear and am afraid to click 'yes'.

**Comment by 50Wliu** [comment]
@JohnLeicht selecting 'yes' will allow Atom to open URL links that start with `atom://`.  This can be used, for example, to link directly to an Atom package or open Atom to a specific file.

**Comment by BinaryMuse** [comment]
@simurai I agree that the messaging for this dialog could be much improved. Any thoughts? (Or maybe @jasonrudolph, who is really good at voice in Atom)

**Comment by jasonrudolph** [comment]
@carpben @JohnLeicht: Thanks for caring about Atom's UX. :heart_eyes:

I found myself wanting to see how other apps handle this situation. Below are examples from various apps and operating systems. I'll follow up shortly with some thoughts on how we might improve Atom's UX in this area.

## Related examples

### Chrome opening alternative protocol

<img width="1000" alt="screen shot 2018-01-24 at 09 44 04 am" src="https://user-images.githubusercontent.com/2988/35340197-77584b78-00f0-11e8-9159-61c828283524.png">

### Safari opening alternative protocol

![demo](https://user-images.githubusercontent.com/2988/35340196-774cf52a-00f0-11e8-9128-c7b839c8251e.gif)

### Windows 10 settings for associating protocols with apps

Step 1
<img width="823" alt="screen_shot_2018-01-24_at_09_59_40_am" src="https://user-images.githubusercontent.com/2988/35340173-6d0200a6-00f0-11e8-97e7-3ad0275c6dc5.png">

Step 2
<img width="872" alt="screen_shot_2018-01-24_at_10_00_17_am" src="https://user-images.githubusercontent.com/2988/35340174-6d0f6a2a-00f0-11e8-81fb-9d235132018f.png">

Step 3
<img width="588" alt="screen shot 2018-01-24 at 10 00 37 am" src="https://user-images.githubusercontent.com/2988/35340201-77717f30-00f0-11e8-8a81-26a3d802ba96.png">

### Windows settings for associating file type with app

![set-associations-900x600](https://user-images.githubusercontent.com/2988/35340202-77c39950-00f0-11e8-909e-b3245a38407b.png)

### macOS setting for associating file type with app

<img width="259" alt="screen shot 2018-01-24 at 09 55 15 am" src="https://user-images.githubusercontent.com/2988/35340198-77640f1c-00f0-11e8-99d6-85b0a4c2522b.png">

### Firefox settings for opening file types

![2015-05-14-07-04-09-28d850](https://user-images.githubusercontent.com/2988/35340195-7721fbe0-00f0-11e8-9e25-8b6243df0eaf.png)

### Little Snitch asking whether you want to permit a network request

![screen shot 2018-01-24 at 10 38 11 am](https://user-images.githubusercontent.com/2988/35341109-e59267c0-00f2-11e8-90ef-1fe21d32d54c.png)

**Comment by jasonrudolph** [comment]
@binarymuse: A couple questions for you:

1. I _think_ I understand what each button does in [the current dialog](https://user-images.githubusercontent.com/13496178/34385155-62980656-eb29-11e7-98e3-9eb85af46cb9.png), but I don't want to make any assumptions. :innocent: For the sake of this discussion, would you be willing to describe the behavior of each button in a few words?

2. I wonder if we can make the dialog more approachable by deemphasizing the act of registering a "protocol handler" and instead speaking in terms of things we do every day (e.g., opening URLs or opening links). What would you think of something like this?

    > **Open this URL in atom?**
    >
    >   `atom://bacon-ipsum-dolor-sit-amet-shoulder-fatback-venison`
    >
    > [ ] Always open atom:// URLs in Atom
    >
    > [Open] [Cancel]

**Comment by simurai** [comment]
> though this prompt is shown at Atom startup, not when a link is clicked (the user must click "Yes" before any links will trigger any action in Atom).

That's a good point. Would it be possible to show @jasonrudolph's [suggested dialog](https://github.com/atom/atom/issues/16453#issuecomment-360177634) only after clicking an `atom://` link for the first time? Then it's easier to understand the dialog's meaning and also shows people how handy this feature is. Maybe like this:

1. When installing Atom, do **Yes, Always** as default **without** asking.
2. Then when a user clicks on an `atom://` link for the **first time**, show the dialog.
3. Now the user can still `Cancel` this one time or opt-out entirely by unchecking `Always open..` + click on `Cancel`.

Yeah, this auto opt-in might be a bit aggressive and in the past we got a huge backlash for the metrics, but I think in this case it's less controversial and in the best interest of most users.

**Comment by JohnLeicht** [comment]
My biggest problem with the dialog was "URI" - which I understand after researching. The term is more encompassing - but, less familiar. A simple example: "atom://" - which is a URL - should resolve the issue. Maybe also mention a typical scenario that this protocol would use.

**Comment by carpben** [comment]
Thanks for taking this seriously. I'm a novice programmer, and I found the message to unclear. I think that good phrasing, and a simple explenation, either in the alert window, or in a link from the alert window should be sufficient. Here are a few suggestions: 

- make it clear that the message is from Atom/Atom Creators, and not from a third organization. 

- Explain what it is used for. 

- If possible, let the user know that it does not risk the security or privacy of the user. 

- Give a recommendation.

**Comment by Teatoller** [comment]
I find it distracting each time it pops up.
I am always waiting for that next time to dig deeper for a better understanding. That is how I landed on this thread.

**Comment by allamiro** [comment]
first thing I searched on what  does that means from security stand point landed to this thread its important to understand what is happening around your system

**Comment by Joshua-Inman** [comment]
Thank you!

**Comment by garrablanca** [comment]
It would be really nice to have in that prompt window a link to know more about the atom:// URIs.

As simple as they just add a "note: Want to know what atom://URIs does? click here!" 😄

**Comment by h3298** [comment]
I'm not afraid of granting this permission, but I did wonder which application you were stealing the default away from.

**Comment by jbarchuk** [comment]
> ...wonder which application you were stealing the default away from.

There's a control panel pic above, here's the whole trail.

Control Panel - Programs - Default Programs - Associate a file type... - Default Apps - (scroll down) - Choose Default Apps by Protocol.

Mine has no default for atom, so it won't interfere with anything.

My main concern is atom protecting itself. I can imagine random atom://my-ATTACK-file text out there.

Personally I don't want it to manage my file EDITING software so closely. That's not necessary. But I'll try it anyway.
